### PR TITLE
Beta register

### PR DESCRIPTION
**Pull request by wkillerud**
Add configuration to filter certain podlet requests by deviceType header. Typically to skip header and footer in certain scenarios.

### CODE REVIEW

File: lib/resource.js [hunk 1]
```diff
@@ -57,19 +57,71 @@ export default class PodiumClientResource {
...
+                        headers: {},
+                        content: '',
+                        css: [],
+                        js: [],
+                        redirect: '',
+                    });
+                }
+            }
+        }
+
```
Comment (digitalsadhu) [inline lib/resource.js:115]: should includeBy take precedence over excludeBy? Should there be any validation of cases? Eg. If you defined the same device type in both excludeBy and includeBy. Should one be able to use both excludeBy and includeBy or should it be if/else? ie. Either you use a black list or a white list but not both.

**Review by digitalsadhu — COMMENTED** [review]
some thoughts/questions...

File: lib/resource.js (hunk 1 above)
Comment (wkillerud) [reply lib/resource.js:115]: Good questions, what do you think?

I'll admit we added `includeBy` mostly for completeness. `excludeBy` is the main case here, so if it opens a whole can of worms maybe we just drop `includeBy` 😅

File: lib/resource.js [hunk 2]
```diff
@@ -57,19 +57,71 @@ export default class PodiumClientResource {
...
+            throw new TypeError(
+                'you must pass an instance of "HttpIncoming" as the first argument to the .fetch() method',
+            );
         const outgoing = new HttpOutgoing(this.#options, reqOptions, incoming);
 
+        if (this.#options.excludeBy) {
+            /**
+             * @type {string[] | undefined}
+             */
+            const exclucedDeviceTypes = this.#options.excludeBy.deviceType;
```
[the remark this answers is not in the document]
Comment (wkillerud) [reply lib/resource.js:70]: Thanks, fixed in https://github.com/podium-lib/client/commit/75dd26275b864357199fa5fce4f5727c3ee14132

**Pull request merged**